music_assistant/providers/opensubsonic/sonic_provider.py

```diff
@@ -330,7 +333,14 @@ async def search(
     async def get_library_artists(self) -> AsyncGenerator[Artist, None]:
         """Provide a generator for reading all artists."""
         artists = await self._run_async(self._conn.get_artists)
+
+        if not artists.index:
+            return
+
         for index in artists.index:
+            if index.artist:
```

Comment:
This is incorrect, I will fix it when I get home this evening, should be `if not index.artist:`

Reply:
Or I can fix it now, nm